cmd/fs-v1.go

```diff
@@ -254,7 +254,11 @@ func newFSObjectLayer(fsPath string) (ObjectLayer, error) {
 
 	// Initialize `format.json`.
 	if err = initFormatFS(fsPath, fsUUID); err != nil {
-		return nil, err
+		// The follow error would occur if there is a conflicting lock held.
+		// we should simply proceed to attempt a read lock.
+		if errorCause(err) != lock.ErrLocked {
+			return nil, err
+		}
```

Comment:
If we proceed to acquire read lock, we would still have format.json at version 1 but the new binary would be storing objects uploaded via it in the new directory structure causing requests routed to the older server to fail. This should not be allowed. Instead, m2 should exit printing a message that older version may be running and exit. To make certain that we were conflicted only by older version we need https://github.com/minio/minio/pull/4512, which proceeds to migration only if we still have older version of servers running.